pkgs/by-name/fo/fortune/package.nix

```diff
@@ -20,11 +21,16 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-Hzh4dyVOleq2H5NyV7QmCfKbmU7wVxUxZVu/w6KsdKw=";
   };
 
-  nativeBuildInputs = [
-    cmake
-    perl
-    rinutils
-  ];
+  nativeBuildInputs =
+    [
+      cmake
+      perl
+      rinutils
+    ]
+    ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+      # "strfile" must be in PATH for cross-compiling builds.
+      fortune
```

Comment:
I'm not  be familiar with cross building. 
Could you tell me why it is necessary to depend on itself? 
And cannot we use `finalAttrs` with `stdenv.mkDerivation` here?

Reply:
It is needed because the `fortune` build compiles and uses a custom `strfile` tool. So we need to get this tool from the native `fortune` package to cross-compile. (The cross build will contain `strfile` for the host platform only.)

I don't think it would make sense to use `finalAttrs` (`finalPackage`) here but I am not sure. Probably you would run into infinite recursion.